src/preferences/dialog/dlgprefsound.cpp

```diff
@@ -605,30 +604,44 @@ void DlgPrefSound::engineClockChanged(int index) {
 // but they'll be close).
 void DlgPrefSound::updateAudioBufferSizes(int sampleRateIndex) {
-    double sampleRate = sampleRateComboBox->itemData(sampleRateIndex).toDouble();
-    int oldSizeIndex = audioBufferComboBox->currentIndex();
-    unsigned int framesPerBuffer = 1; // start this at 0 and inf loop happens
-    // we don't want to display any sub-1ms buffer sizes (well maybe we do but I
-    // don't right now!), so we iterate over all the buffer sizes until we
-    // find the first that gives us a buffer size >= 1 ms -- bkgood
-    // no div-by-0 in the next line because we don't allow srates of 0 in our
-    // srate list when we construct it in the ctor -- bkgood
-    for (; framesPerBuffer / sampleRate * 1000 < 1.0; framesPerBuffer *= 2) {
-    }
+    QVariant oldSizeIndex = audioBufferComboBox->currentData();
     audioBufferComboBox->clear();
-    for (unsigned int i = 0; i < SoundManagerConfig::kMaxAudioBufferSizeIndex; ++i) {
-        const auto latency = static_cast<float>(framesPerBuffer / sampleRate * 1000);
-        // i + 1 in the next line is a latency index as described in SSConfig
-        audioBufferComboBox->addItem(tr("%1 ms").arg(latency,0,'g',3), i + 1);
-        framesPerBuffer <<= 1; // *= 2
-    }
-    if (oldSizeIndex < audioBufferComboBox->count() && oldSizeIndex >= 0) {
-        audioBufferComboBox->setCurrentIndex(oldSizeIndex);
+    if (m_config.getAPI() == MIXXX_PORTAUDIO_JACK_STRING) {
+        // in case of jack we configure the frames/period
+        // we cannot calc the resulting buffer size in ms because the
+        // Sample rate is not known yet. We assume 48000 KHz here
+        // to calculate the buffer size index
+        audioBufferComboBox->addItem(tr("auto (<= 1024 frames/period)"), 5);
+        audioBufferComboBox->addItem(tr("2048 frames/period"), 6);
+        audioBufferComboBox->addItem(tr("4096 frames/period"), 7);
     } else {
-        // set it to the max, let the user dig if they need better latency. better
-        // than having a user get the pops on first use and thinking poorly of mixxx
-        // because of it -- bkgood
-        audioBufferComboBox->setCurrentIndex(audioBufferComboBox->count() - 1);
+        double sampleRate = sampleRateComboBox->itemData(sampleRateIndex).toDouble();
+        unsigned int framesPerBuffer = 1; // start this at 0 and inf loop happens
+        // we don't want to display any sub-1ms buffer sizes (well maybe we do but I
+        // don't right now!), so we iterate over all the buffer sizes until we
+        // find the first that gives us a buffer size >= 1 ms -- bkgood
+        // no div-by-0 in the next line because we don't allow srates of 0 in our
+        // srate list when we construct it in the ctor -- bkgood
+        for (; framesPerBuffer / sampleRate * 1000 < 1.0; framesPerBuffer *= 2) {
+        }
+        for (unsigned int i = 0; i < SoundManagerConfig::kMaxAudioBufferSizeIndex; ++i) {
```

Comment:
```suggestion
        for(const auto& i : SoundManagerConfig::AudioBufferSizeIndex ) {
```

Reply:
This does not work in c++, without a lot of boilerplate code.

Reply:
Did you tried it?

Reply:
I just did it out of carious:
```
[5/10] Building CXX object CMakeFiles/mixxx-lib.dir/src/preferences/dialog/dlgprefsound.cpp.o
FAILED: CMakeFiles/mixxx-lib.dir/src/preferences/dialog/dlgprefsound.cpp.o 
ccache /usr/bin/c++  -DHAVE_INET_ATON -DHAVE_INET_PTON -DHAVE_UNISTD_H -DMIXXX_BUILD_DEBUG -DMIXXX_DEBUG_ASSERTIONS_ENABLED -DQT_CONCURRENT_LIB -DQT_CORE_LIB -DQT_DBUS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_OPENGL_LIB -DQT_SCRIPTTOOLS_LIB -DQT_SCRIPT_LIB -DQT_SQL_LIB -DQT_SVG_LIB -DQT_TABLET_SUPPORT -DQT_TESTCASE_BUILDDIR=\"/home/daniel/workspace/2.3/build/Debug\" -DQT_TESTLIB_LIB -DQT_USE_QSTRINGBUILDER -DQT_WIDGETS_LIB -DQT_X11EXTRAS_LIB -DQT_XML_LIB -DSFC_SUPPORTS_SET_COMPRESSION_LEVEL -D__BATTERY__ -D__BROADCAST__ -D__BULK__ -D__FFMPEG__ -D__HID__ -D__KEYFINDER__ -D__LILV__ -D__LINUX__ -D__MAD__ -D__MODPLUG__ -D__OPUS__ -D__QTKEYCHAIN__ -D__SNDFILE__ -D__SQLITE3__ -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -D__UNIX__ -D__VINYLCONTROL__ -D__WV__ -Dx86_64 -Imixxx-lib_autogen/include -I../../src -Isrc -isystem ../../lib/fidlib -isystem ../../lib/googletest/googletest/include -isystem ../../lib/portaudio -isystem ../../lib/rigtorp/SPSCQueue/include -isystem ../../lib/qtscript-bytearray -isystem ../../lib/replaygain -isystem ../../lib/reverb -isystem /usr/include/glib-2.0 -isystem /usr/lib/x86_64-linux-gnu/glib-2.0/include -isystem ../../lib/libshout-idjc/include -isystem /usr/include/opus -isystem lib/keyfinder-install/include -isystem ../../lib/kaitai -isystem ../../lib/mp3guessenc-0.27.4 -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtConcurrent -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -isystem /usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -isystem /usr/include/x86_64-linux-gnu/qt5/QtGui -isystem /usr/include/x86_64-linux-gnu/qt5/QtNetwork -isystem /usr/include/x86_64-linux-gnu/qt5/QtOpenGL -isystem /usr/include/x86_64-linux-gnu/qt5/QtWidgets -isystem /usr/include/x86_64-linux-gnu/qt5/QtScript -isystem /usr/include/x86_64-linux-gnu/qt5/QtScriptTools -isystem /usr/include/x86_64-linux-gnu/qt5/QtSql -isystem /usr/include/x86_64-linux-gnu/qt5/QtSvg -isystem /usr/include/x86_64-linux-gnu/qt5/QtTest -isystem /usr/include/x86_64-linux-gnu/qt5/QtXml -isystem /usr/include/x86_64-linux-gnu/qt5/QtX11Extras -isystem /usr/include/x86_64-linux-gnu/qt5/QtDBus -isystem ../../lib/qm-dsp -isystem ../../lib/qm-dsp/include -isystem /usr/include/taglib -isystem /usr/include/libupower-glib -isystem /usr/include/lilv-0 -isystem ../../lib/hidapi/hidapi -isystem /usr/include/libusb-1.0 -isystem ../../lib/xwax -isystem /usr/include/wavpack -fdiagnostics-color=auto -g   -pipe -O3 -ffast-math -funroll-loops -fomit-frame-pointer -mtune=generic -Wall -Wextra -Woverloaded-virtual -Wfloat-conversion -Werror=return-type -fPIC -pthread -Wno-unused-parameter -std=gnu++17 -MD -MT CMakeFiles/mixxx-lib.dir/src/preferences/dialog/dlgprefsound.cpp.o -MF CMakeFiles/mixxx-lib.dir/src/preferences/dialog/dlgprefsound.cpp.o.d -o CMakeFiles/mixxx-lib.dir/src/preferences/dialog/dlgprefsound.cpp.o -c ../../src/preferences/dialog/dlgprefsound.cpp
../../src/preferences/dialog/dlgprefsound.cpp: In member function ‘void DlgPrefSound::updateAudioBufferSizes(int)’:
../../src/preferences/dialog/dlgprefsound.cpp:627:70: error: expected primary-expression before ‘)’ token
  627 |         for(const auto& i : SoundManagerConfig::AudioBufferSizeIndex ) {
      |                                                                      ^
ninja: build stopped: subcommand failed.
```

Reply:
Ok